File: internal/domain/central_info.go
package domain

type CentralInfo struct {
	GlobalVariables  []EnvVariable     `json:"globalVariables"`
	ApplicationInfos []ApplicationInfo `json:"applicationInfos"`
}

type EnvVariable struct {
	Name     string `json:"name"`
	Value    string `json:"value"`
	IsActive bool   `json:"isActive"`
}

type ApplicationInfo struct {
	AppName      string        `json:"appName"`
	EnvVariables []EnvVariable `json:"envVariables"`
	AppArguments []string      `json:"appArguments"`
	BaseDir      string        `json:"baseDir"`
	JarPath      string        `json:"jarPath"`
	StartOrder   uint8         `json:"startOrder"`
	IsActive     bool          `json:"isActive"`
	HasGit       bool          `json:"hasGit"`
	HasMaven     bool          `json:"hasMaven"`
}

File: frontend/src/app/model/central-info.ts
import {EnvVariable} from './env-variable';
import {ApplicationInfo} from './application-info';

export class CentralInfo {
  globalVariables: EnvVariable[];
  applicationInfos: ApplicationInfo[];


  constructor() {
    this.globalVariables = [];
    this.applicationInfos = [];
  }
}
